git-https-proxy/mitmproxy.js

```diff
@@ -0,0 +1,77 @@
+
+/*!
+ * Copyright 2020 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const proxy = require('http-mitm-proxy')();
+const url = require('url');
+
+const proxyPort = process.env.MITM_PROXY_PORT || 8080;
+const gitlabOauthToken = process.env.GITLAB_OAUTH_TOKEN;
+const encodedCredentials = Buffer.from(
+  `oauth2:${gitlabOauthToken}`
+).toString('base64');
+const repoUrl = new url.URL(process.env.REPOSITORY_URL);
+
+let defaultPort;
+switch (repoUrl.protocol) {
+  case 'https:':
+    defaultPort = 443;
+  case 'http:':
+    defaultport = 80;
+}
```

Comment:
Nothing wrong here, but I guess you want to always set a `default` case.
I would simplify a bit the syntax if `80` is also the default port:

```
const defaultPort = repoUrl.protocol === 'https' ?
  443 :
  80;
```

Reply:
Good point - added the `default:` case. However, I prefer to keep it undefined if the protocol is not explicitly set to 'http' or 'https' (which should never happen, but still).